features/netsocket/nsapi_types.h

```diff
@@ -24,27 +24,42 @@ extern "C" {
 #endif
 
 
-/** Enum of standardized error codes 
+/** Enum of standardized error codes
  *
  *  Valid error codes have negative values and may
  *  be returned by any network operation.
  *
  *  @enum nsapi_error_t
  */
 typedef enum nsapi_error {
+    NSAPI_ERROR_OK            =  0,        /*!< no error */
     NSAPI_ERROR_WOULD_BLOCK   = -3001,     /*!< no data is not available but call is non-blocking */
     NSAPI_ERROR_UNSUPPORTED   = -3002,     /*!< unsupported functionality */
     NSAPI_ERROR_PARAMETER     = -3003,     /*!< invalid configuration */
     NSAPI_ERROR_NO_CONNECTION = -3004,     /*!< not connected to a network */
     NSAPI_ERROR_NO_SOCKET     = -3005,     /*!< socket not available for use */
     NSAPI_ERROR_NO_ADDRESS    = -3006,     /*!< IP address is not known */
     NSAPI_ERROR_NO_MEMORY     = -3007,     /*!< memory resource not available */
-    NSAPI_ERROR_DNS_FAILURE   = -3008,     /*!< DNS failed to complete successfully */
-    NSAPI_ERROR_DHCP_FAILURE  = -3009,     /*!< DHCP failed to complete successfully */
-    NSAPI_ERROR_AUTH_FAILURE  = -3010,     /*!< connection to access point faield */
-    NSAPI_ERROR_DEVICE_ERROR  = -3011,     /*!< failure interfacing with the network procesor */
+    NSAPI_ERROR_NO_SSID       = -3008,     /*!< ssid not found */
+    NSAPI_ERROR_DNS_FAILURE   = -3009,     /*!< DNS failed to complete successfully */
+    NSAPI_ERROR_DHCP_FAILURE  = -3010,     /*!< DHCP failed to complete successfully */
+    NSAPI_ERROR_AUTH_FAILURE  = -3011,     /*!< connection to access point failed */
+    NSAPI_ERROR_DEVICE_ERROR  = -3012,     /*!< failure interfacing with the network processor */
 } nsapi_error_t;
 
+/** Enum of encryption types
+ *
+ *  The security type specifies a particular security to use when
+ *  connected to a WiFi network
+ */
+typedef enum nsapi_security {
+    NSAPI_SECURITY_NONE         = 0x0,      /*!< open access point */
+    NSAPI_SECURITY_WEP          = 0x1,      /*!< phrase conforms to WEP */
+    NSAPI_SECURITY_WPA          = 0x2,      /*!< phrase conforms to WPA */
+    NSAPI_SECURITY_WPA2         = 0x3,      /*!< phrase conforms to WPA2 */
+    NSAPI_SECURITY_WPA_WPA2     = 0x4,      /*!< phrase conforms to WPA/WPA2 */
+    NSAPI_SECURITY_UNSSUPPORTED = 0xFF,     /*!< unknown/unsupported security in scan results */
```

Comment:
We're talking about scan results only here. If you try to connect to UNSUPPORTED network it would fail.